Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt

```diff
@@ -34,18 +38,39 @@ class Surveys @Inject constructor(
     }
 
     suspend fun requestDetails(type: Type): Survey {
+        val surveyConfig = appConfigProvider.getAppConfig().survey
+        Timber.v("Requested survey: %s", surveyConfig)
+        /* no check here:
+         * if surveyOnHighRisk is not enabled, this use case shouldn't have been started in the first place
+         */
+//        if (type == Type.HIGH_RISK_ENCOUNTER && !surveyConfig.surveyOnHighRiskEnabled) {
+//            throw SurveyException(SurveyException.Type.HIGH_RISK_NOT_ENABLED)
+//        }
+        otpRepo.otpAuthorizationResult?.apply {
+            if (authorized && redeemedAt.toDateTime().monthOfYear() == Instant.now().toDateTime().monthOfYear()) {
```

Comment:
Are you sure that we need to check if it's the same month? For me, it would make more sense to check whether `redeemedAt` is older than one month?

Reply:
we are supposed to compare the cal month